### PR TITLE
UI: Add about dialog

### PR DESCRIPTION
**Pull request by cg2121**
This adds an about dialog to the help menu.

I separated the more controversial changes into different commits, so they can be easily accepted or rejected. 

screenshot from 2018-08-19 00-30-48
screenshot from 2018-08-19 00-31-19

### CODE REVIEW

File: UI/data/locale/en-US.ini [hunk 1]
```diff
@@ -857,3 +858,12 @@ CustomColor="Custom Color"
 
 # Global settings for the browser source
 BrowserSource.EnableHardwareAcceleration="Enable Browser Source Hardware Acceleration"
+
+#About dialog
```
Comment (RytoEX) [inline UI/data/locale/en-US.ini]: There should be a space between the # and the proceeding text.

File: UI/window-basic-about.cpp [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+OBSAbout::OBSAbout(QWidget *parent)
+	: QDialog(parent),
+	  ui(new Ui::OBSAbout)
+{
+	ui->setupUi(this);
+
+	QString bitness;
+
+	if(sizeof(void*) == 4) {
```
Comment (RytoEX) [inline UI/window-basic-about.cpp]: Style:  You must have a space between `if` and the opening parenthesis.

Comment (RytoEX) [inline UI/window-basic-about.cpp]: Style:  This `if` block does not need curly braces.

File: UI/window-basic-about.cpp [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+		bitness = " (32 bit)";
+	} else if(sizeof(void*) == 8) {
+		bitness = " (64 bit)";
+	}
+
+	ui->version->setText(QString::number(LIBOBS_API_MAJOR_VER) + "." +
+		QString::number(LIBOBS_API_MINOR_VER) + "." +
```
Comment (RytoEX) [inline UI/window-basic-about.cpp]: Style:  Indent twice for a line continuation.

**Comment by SuslikV** [comment]
Easter egg is missing.
The icon is too huge.
"About" link in "About" window is a bit strange thing.

**Comment by RytoEX** [comment]
macOS and Linux builds are still failing on CMakeLists.txt line 158.
```
CMake Error at CMakeLists.txt:158 (configure_file):
  configure_file Problem configuring file
```

**Comment by RytoEX** [comment]
@WizardCM @WizardCM I hadn't caught that.  I agree - removing the version number from the title bar is not a good idea.

File: UI/forms/OBSAbout.ui [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+  </property>
+  <property name="windowTitle">
+   <string>About</string>
+  </property>
+  <widget class="QLabel" name="icon">
+   <property name="geometry">
+    <rect>
+     <x>30</x>
+     <y>30</y>
+     <width>261</width>
```
Comment (DungFu) [inline UI/forms/OBSAbout.ui]: seems like a strange hard coded size for the icon (and maybe a little large?)

Comment (Heufneutje) [reply UI/forms/OBSAbout.ui]: I think having it large is fine. VLC for example does that as well.

**Comment by RytoEX** [comment]
This is _still_ failing on macOS and Linux (Ubuntu 14.04) CI with this CMake line:
```
configure_file(${CMAKE_CURRENT_SOURCE_DIR}/AUTHORS ${CMAKE_INSTALL_PREFIX}/data/obs-studio/authors/AUTHORS COPYONLY)
```

I'm unable to investigate that further, but that's the current major issue here.

**Comment by jp9000** [comment]
This is really cool.  However, it seems incomplete.  Clicking "Authors" causes a "file not found" error, then it tries to close the entire program and crashes.

**Comment by cg2121** [comment]
@jp9000 how are the files moved from the build directory to the actual install directory? That is the thing that I can't figure out.

**Pull request merged**